rules/opentitan_test.bzl

```diff
@@ -297,24 +300,43 @@ def opentitan_functest(
 
     deps = kwargs.pop("deps", [])
     all_tests = []
+    target_params = {}
 
     # Only build SW for devices we are running tests on.
     devices_to_build_for = []
     for target in targets:
         if target == "dv":
             devices_to_build_for.append("sim_dv")
+            target_params["sim_dv"] = dv_params() if not dv else dv
         elif target == "verilator":
             devices_to_build_for.append("sim_verilator")
-        elif target == "cw310":
+            target_params["sim_verilator"] = verilator_params() if not verilator else verilator
+        elif target in ["cw310_rom", "cw310_test_rom"]:
             devices_to_build_for.append("fpga_cw310")
+
+            # Deep-copy the `cw310` dict.
+            cw310_ = cw310_params() if not cw310 else {k: v for k, v in cw310.items()}
+            if "test_rom" in target:
```

Comment:
Nit: I think `if target.endswith("test_rom")` would be clearer. I actually didn't realize that `in` can be used to find substrings!

Reply:
Sure, but I think we should clean up this whole file pretty soon so I'll leave this as is for now.